docs/source/tutorial/extract.rst

```diff
@@ -0,0 +1,432 @@
+
+    source_data_url = 'file://path/to/family_and_phenotype.tsv'
+
+The first thing that the extractor does for every config file is fetch the
+related source data. This specifies where the file lives so that the code can
+fetch it.
+
+Supported protocol prefixes are:
+``file://``, ``s3://``, ``http://``, ``https://``
+
+Loading the data
+----------------
+
+.. code-block:: python
+
+    source_data_loading_parameters = {}
+
+This file contains tab-separated values with a simple layout where the first
+row is the column headers, so it gets loaded correctly by default. With more
+complex files we'd have to define how to load the data. Those arguments would
+go here. For more information on custom loading read <TODO>.
+
+Extract operations
+------------------
+
+The operations list
+^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    operations = [
+        ...
+    ]
+
+This is a list of special functions that implement all of the
+convert/unify/reshape/standardize stuff described earlier. The most useful
+functions are already written for you. You just have to invoke them
+appropriately.
+
+For more information about extract operation functions, read <TODO>.
+
+A value map operation with functional replacements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    value_map(
+        in_col="participant",
+        m={
+            r"PID(\d+)": lambda x: int(x),  # strip PID and 0-padding
+        },
+        out_col=CONCEPT.PARTICIPANT.ID
+    )
+
+This says "Use the ``participant`` column as input (in_col=), replace
+everything that matches the regular expression pattern ``^PID(\d+)$`` with just
+the captured part and remove the zero padding by running the captured part
+through the function ``lambda x: int(x)`` (m={...}), and then output the result
+to a ``CONCEPT.PARTICIPANT.ID`` column (out_col=)."
+
+The resulting intermediate output will look like:
+
+.. csv-table::
+    :header: "index", "<CONCEPT.PARTICIPANT.ID>"
+
+    "0", "1"
+    "1", "2"
+    "2", "3"
+    "3", "4"
+    "4", "5"
+    "5", "6"
+    "6", "7"
+    "7", "8"
+    "8", "9"
+
+``lambda x: int(x)`` could be replaced by just ``int``, since the two
+expressions are functionally equivalent (both single-argument functions that
+effectively strip the leading zeros).
+
+We could also have kept these IDs as they were and instead converted the
+mother/father IDs, but, in the absence of an overriding directive such as input
+from the investigators about their preferences, it doesn't really make a
+difference which way we choose.
+
+A value map operation with variable replacements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    value_map(
+        in_col="gender",
+        m={
+            "F": constants.GENDER.FEMALE,
+            "M": constants.GENDER.MALE
+        },
+        out_col=CONCEPT.PARTICIPANT.GENDER
+    )
+
+This says "Use the ``gender`` column as input (in_col=), replace everything
```

Comment:
Can we mention something about the fact that standard codes are defined in the constants module and you should always try to use them if possible?

Reply:
👌